sw/device/tests/hmac_enc_irq_test.c

```diff
@@ -0,0 +1,147 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_hmac.h"
+#include "sw/device/lib/dif/dif_rv_plic.h"
+#include "sw/device/lib/irq.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/hmac_testutils.h"
+#include "sw/device/lib/testing/rv_plic_testutils.h"
+#include "sw/device/lib/testing/test_framework/ottf.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+static const uint32_t kPlicTarget = kTopEarlgreyPlicTargetIbex0;
+
+dif_hmac_t hmac;
+dif_rv_plic_t plic;
+
+/**
+ * Flags to check whether the interrupts have fired.
+ */
+static bool hmac_empty_irq;
+static bool hmac_done_irq;
```

Comment:
I would specify them as `volitile` to make sure the compiler won't optimize them inside the irq handler.